gobblin-iceberg/src/test/java/org/apache/gobblin/iceberg/writer/HiveMetadataWriterTest.java

```diff
@@ -118,14 +118,14 @@ public class HiveMetadataWriterTest extends HiveMetastoreTest {
   IMetaStoreClient client;
   private static TestHiveMetastore testHiveMetastore;
 
-  @AfterClass
+  @AfterSuite
   public void clean() throws Exception {
-    //Finally stop the metaStore
-    stopMetastore();
     gobblinMCEWriter.close();
     FileUtils.forceDeleteOnExit(tmpDir);
+    //Finally stop the metaStore
+    stopMetastore();
   }
-  @BeforeClass
+  @BeforeSuite
```

Comment:
Before class causes some strange flakiness with when this executes. Before and After Suite is the correct behavior (it acts as the entry point and exit point for all the test code with hive)